src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -228,5 +232,40 @@ private void SendFileInternal(string? fileName, ReadOnlySpan<byte> preBuffer, Re
                 Send(postBuffer);
             }
         }
+
+        internal void DisposeHandle()
+        {
+            _handle.Dispose();
+        }
+
+        internal void ClearHandle()
+        {
+            _handle = null!;
+        }
+
+        internal Socket CopyStateFromSource(Socket source)
+        {
+            _addressFamily = source._addressFamily;
+            _closeTimeout = source._closeTimeout;
+            _disposed = source._disposed;
+            _handle = source._handle;
+            _isConnected = source._isConnected;
+            _isDisconnected = source._isDisconnected;
+            _isListening = source._isListening;
+            _nonBlockingConnectInProgress = source._nonBlockingConnectInProgress;
+            _protocolType = source._protocolType;
+            _receivingPacketInformation = source._receivingPacketInformation;
+            _remoteEndPoint = source._remoteEndPoint;
+            _rightEndPoint = source._rightEndPoint;
+            _socketType = source._socketType;
+            _willBlock = source._willBlock;
+            _willBlockInternal = source._willBlockInternal;
+            _localEndPoint = source._localEndPoint;
+            _multiBufferReceiveEventArgs = source._multiBufferReceiveEventArgs;
+            _multiBufferSendEventArgs = source._multiBufferSendEventArgs;
+            _pendingConnectRightEndPoint = source._pendingConnectRightEndPoint;
+            _singleBufferReceiveEventArgs = source._singleBufferReceiveEventArgs;
+            return this;
```

Comment:
It may make sense to add something like
https://github.com/dotnet/runtime/blob/9a72a29c60f2254a72cd36ee2ef0007dba1fc2d7/src/libraries/Common/src/System/Net/Security/SslClientAuthenticationOptionsExtensions.cs#L32-L33

This would ensure that we update this function if we add new property to `Socket`. This method would be easy to miss IMHO

Reply:
Yeap, I'll add something like this!